src/c++/perf_analyzer/genai-pa/tests/test_cli.py

```diff
@@ -24,14 +24,70 @@
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+from pathlib import Path
+
 import pytest
-from genai_pa.main import run
+from genai_pa import parser
+
+
+class TestCLIArguments:
+    @pytest.mark.parametrize(
+        "arg, expected_output",
+        [
+            (["-h"], "CLI to profile LLMs and Generative AI models with Perf Analyzer"),
+            (
+                ["--help"],
+                "CLI to profile LLMs and Generative AI models with Perf Analyzer",
+            ),
+        ],
+    )
+    def test_help_arguments_output_and_exit(self, arg, expected_output, capsys):
+        with pytest.raises(SystemExit) as excinfo:
+            _ = parser.parse_args(arg)
+
+        # Check that the exit was successful
+        assert excinfo.value.code == 0
+
+        # Capture that the correct message was displayed
+        captured = capsys.readouterr()
+        assert expected_output in captured.out
+
+    @pytest.mark.parametrize(
+        "arg, expected_attributes",
+        [
+            (["-b", "2"], {"batch_size": 2}),
+            (["--batch-size", "2"], {"batch_size": 2}),
+            (["--concurrency", "3"], {"concurrency_range": "3"}),
+            (["--max-threads", "4"], {"max_threads": 4}),
+            (
+                ["--profile-export-file", "text.txt"],
+                {"profile_export_file": Path("text.txt")},
+            ),
+            (["--request-rate", "1.5"], {"request_rate_range": "1.5"}),
+            (["--service-kind", "triton"], {"service_kind": "triton"}),
+            (["--service-kind", "openai"], {"service_kind": "openai"}),
+            # TODO: Remove streaming from implementation. It is invalid with HTTP.
+            # (["--streaming"], {"streaming": True}),
+            (["--version"], {"version": True}),
+            (["-u", "test_url"], {"u": "test_url"}),
+            (["--url", "test_url"], {"u": "test_url"}),
+        ],
+    )
+    def test_arguments_output(self, arg, expected_attributes, capsys):
+        combined_args = ["--model", "test_model"] + arg
+        args = parser.parse_args(combined_args)
+
+        # Check that the attributes are set correctly
+        for key, value in expected_attributes.items():
+            assert getattr(args, key) == value
```

Comment:
note: I think some of the expected attributes may change if Elias uses `dest` like `parser.add_argument("--batch-size", ..., dest="b")` in a later PR, so these parameters may have to change to `{"b": 2}` etc.

Reply:
I tried to guard against that with metavar but its a good call out.

Reply:
AFAIK, `metavar` only modifies the `--help` output, but won't change the attributes:
```python
import argparse
parser = argparse.ArgumentParser()
parser.add_argument("-b", "--batch-size", default=1, dest="b", metavar="batch_size")
args = parser.parse_args()

for attr in ["b", "batch_size"]:
    print(f'{attr=}: {hasattr(args, attr)=}')
```
```bash
$ python test.py 
attr='b': hasattr(args, attr)=True
attr='batch_size': hasattr(args, attr)=False
```

Reply:
Thanks for calling this out and expounding on how metavar works, Ryan! We'll have these tests now, so we should make sure they are still passing after any implementation changes. If not, we can update the tests to match changes in how we want to the CLI designed.

Reply:
I think it's moot since it looks like the current code isn't using `metavar` for `--batch-size` and is instead just mapping it later on before calling subprocess. Just mentioning it in case you got unexpected test failures like this later after a `dest` change.